New Token Dashboard

edspencer opened the pull request:
Addresses https://github.com/oplabs/launchpad/issues/3984 (see [UX designs](https://www.figma.com/file/UYDaza4snnYYSdnf57CVYY/Story-Website?node-id=0%3A1&t=jreoMRcY00uFhpG7-0))

I've tried to follow existing conventions but may have missed a trick or two. I introduced some new tailwind colors for those blue backgrounds - seemed a reasonable way but I didn't see any other colors there so thought I'd mention it. 

The new page seems to behave itself reasonably at all screen sizes, though it's not pixel perfect to the design - in general I've favored well-behaving responsive behavior over a variety of screen resolutions vs getting everything pixel perfect on the two screen widths in figma.

Commit:
WIP on token dashboard

Commit:
nice progress

Commit:
better responsive design stuff

Commit:
hooked up most of the remaining data

Commit:
refactor into separate components

Commit:
alchemy

Commit:
fetch staked ogn from API

Commit:
little fixes

nickick commented on src/styles/zglobals.css:
```diff
@@ -0,0 +1,22 @@
+/**
+ * And why is this file now called zglobals.css? Because of https://github.com/vercel/next.js/issues/16630
+ * Some files, including _app.tsx, import both this and origin-storybook/lib/styles.css,
+ * but the order of inclusion matters because we're loading some Tailwind twice for some reason and
```
Yep we ended up including Tailwind twice because the storybook components come with Tailwind styling and this app also uses Tailwind, not sure how to more cleanly have both sets without loading them both in.

edspencer replied:
It seems like we only get a subset of Tailwind along with the storybook components, so seems the second include is needed to fill in the gaps. I think we probably can't just bundle the whole of Tailwind in storybook for a bunch of reasons, but we may be able to configure this repo's Tailwind inclusion to avoid bringing in the declarations already brought in by storybook. That seems reasonably clean, but is probably a P3 as this seems to work for now.

nickick commented on src/styles/zglobals.css:
```diff
@@ -0,0 +1,22 @@
+ * occasionally it causes a weird bug where the styles are not applied. The only way to make this file
+ * appear after origin-storybook/lib/styles.css is to make its filename >= style.css. Hence z.
+ * The specific issue it caused was making `w-full sm:w-auto` not work as `w-full` is declared in both
+ * .css files, and was dominating the `sm:w-auto` declaration (which only exists once). This is probably a bug.
```
I had noticed issues like this and didn't know what was going on. Glad you figured out the issue.

nickick replied:
Didn't realize declaring css classes multiple times increased specificity, TIL!

nickick commented on pages/dashboard.tsx:
```diff
@@ -0,0 +1,95 @@
+}) => {
+  //FIXME: add actual implementation here
+  const rewardsInfo = {
+    ethInRewardsPool: 53.6554,
```
looks like this remains to be implemented, did we want to have that in this PR or a future one?

nickick replied:
looks like this is fixed :)

Commit:
final data integration

edspencer commented:
One slightly sad thing is that the final row of exchanges here isn't centered:

![image](https://user-images.githubusercontent.com/6929/218870321-a5354c94-1602-4f16-9e9e-ccdaeb754f7a.png)

In the Figma those 2 on the bottom row are `justify-evenly` or similar, but in the implementation I used a CSS grid, which didn't seem to let me do that. I didn't have much luck with using flex due to the interactions with the SVGs in each grid cell - they just weren't auto-sizing properly for some reason. If you know a way to make that better (and it's not hours of work) then I'm all ears.

Commit:
nitpick

nickick commented on src/components/Dashboard.tsx:
```diff
@@ -1,106 +1,176 @@
-import Image from 'next/future/image'
-import styles from '../styles/Home.module.css'
-import { formatCurrency } from '../helpers/math'
+import {
```
these are all just prettier formatting changes, right?

edspencer replied:
I think so, yea. I guess I made some changes to this file at first before realizing I was in the wrong place and undoing it all, but the formatting gets applied each time I hit save.

nickick commented:
I'm noticing a bottom white strip below the excluded wallets section, I think we don't want that, right?

![Screenshot 2023-02-14 at 6 57 19 PM](https://user-images.githubusercontent.com/1316662/218890393-4643c689-2d27-40c2-8258-25a9a0fc4727.png)

nickick commented:
> One slightly sad thing is that the final row of exchanges here isn't centered:
> 
> ![image](https://user-images.githubusercontent.com/6929/218870321-a5354c94-1602-4f16-9e9e-ccdaeb754f7a.png)
> 
> In the Figma those 2 on the bottom row are `justify-evenly` or similar, but in the implementation I used a CSS grid, which didn't seem to let me do that. I didn't have much luck with using flex due to the interactions with the SVGs in each grid cell - they just weren't auto-sizing properly for some reason. If you know a way to make that better (and it's not hours of work) then I'm all ears.

I'm not sure either on how I would get the bottom to center evenly, but this seems like a small overall issue and the tradeoff for having auto-resizing SVG logos seems worth it to me.

nickick commented on src/components/dashboard/StakeCTA.tsx:
```diff
@@ -0,0 +1,102 @@
+          gradients={["#AFA1FF", "#F68BFF", "#FD9AC2"]}
+        />
+      </Typography.H2>
+      <Typography.H6 className="my-8 text-white opacity-75 ">
```
nit: space before double-quote at the end of this line

edspencer replied:
fixed :)

nickick commented on pages/dashboard.tsx:
```diff
@@ -0,0 +1,92 @@
+  const [ognInfo, seoRes, walletBalances, stakingData] = await Promise.all([
+    requestOgnData(),
+    fetchAPI("/story/page/en/%2F"),
+    requestGovernanceWalletBalances(),
```
we'll need to add the Alchemy key to the Heroku env configs before we deploy this, right?

edspencer replied:
Yes - I don't see it on https://dashboard.heroku.com/teams/originprotocol/apps though... thought I'd have permission but seems maybe I don't. Can you add that or point me in the right direction?

It ~~will~~ should fail gracefully without the env var set, but some data won't populate.

nickick replied:
Ah yeah it's in a different Heroku group.

@franckc can you give Ed access to the Frontend Heroku teams? https://dashboard.heroku.com/teams/frontends/apps

nickick reviewed (approved):
LGTM, aside from 1 small styling issue at the bottom of the excluded wallets section. Nice work!

Commit:
Fix white background on ExcludedWallets

Commit:
nit

edspencer commented:
> LGTM, aside from 1 small styling issue at the bottom of the excluded wallets section. Nice work!

Thanks, good catch on the ExcludedWallets whitespace - I've fixed that and think I've addressed your other comments. I think this one's good to merge now, but we do need to add the heroku env var before deploying.
